### PR TITLE
compute: rename s3_oneshot sink to copy_to_s3_oneshot

### PR DESCRIPTION
**Pull request by moulimukherjee**
Follow up from https://github.com/MaterializeInc/materialize/pull/25042, renaming the `s3_oneshot` sink to `copy_to_s3_oneshot` to make the naming consistent with `CopyTo` prefix used elsewhere.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - 

cc @teskje

### CODE REVIEW

File: src/compute-types/src/sinks.proto [hunk 1]
```diff
@@ -33,7 +36,7 @@ message ProtoComputeSinkConnection {
     oneof kind {
         google.protobuf.Empty subscribe = 1;
         ProtoPersistSinkConnection persist = 2;
-        ProtoS3OneshotSinkConnection s3_oneshot = 3;
+        ProtoS3OneshotSinkConnection copy_to_s3_oneshot = 3;
```
Comment (teskje) [inline src/compute-types/src/sinks.proto:39]: We should also rename the `Proto` type!

Comment (moulimukherjee) [reply src/compute-types/src/sinks.proto:39]: yes 🙈, updating

Comment (moulimukherjee) [reply src/compute-types/src/sinks.proto]: done!

**Review by teskje — APPROVED** [review]
LGTM!

**Pull request merged**